client/src/components/Indices/IndexFilter.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+    <span>
+        <b-input-group>
+            <DebouncedInput :delay="debounceDelay" v-slot="{ value, input }" v-model="localFilter">
+                <b-form-input
+                    :id="id"
+                    name="query"
+                    :value="value"
+                    autocomplete="off"
+                    :placeholder="placeholder | localize"
+                    data-description="filter index input"
+                    class="search-query"
+                    :size="size"
+                    @input="input"
+                    @keyup.esc="onReset" />
+            </DebouncedInput>
+            <b-input-group-append>
+                <b-button
+                    data-description="show deleted filter toggle"
+                    @click="onHelp"
+                    title="Advanced Filtering Help"
+                    :size="size">
+                    <icon icon="question" />
+                </b-button>
+            </b-input-group-append>
+        </b-input-group>
+        <b-modal v-model="showHelp" title="Filtering Options Help" ok-only>
+            <div v-html="helpHtml"></div>
+        </b-modal>
+    </span>
+</template>
+
+<script>
+import DebouncedInput from "components/DebouncedInput";
+
+import { BInputGroup, BInputGroupAppend, BButton, BModal } from "bootstrap-vue";
+
+/**
+ * Component for the search/filter button on the top of Galaxy object index grids.
+ */
+export default {
+    components: {
+        DebouncedInput,
+        BInputGroup,
+        BInputGroupAppend,
+        BButton,
+        BModal,
+    },
+    props: {
+        id: {
+            type: String,
+            required: true,
+        },
+        placeholder: {
+            type: String,
+            required: true,
+        },
+        helpHtml: {
+            type: String,
+            required: true,
+        },
+        debounceDelay: {
+            type: Number,
+            default: 500,
+        },
+        size: {
+            type: String,
+            value: "sm",
+        },
+        value: {},
```

Comment:
Can this specify a type and default or required?

Reply:
Done!